packages/admin/blocks-admin/src/blocks/factories/createOneOfBlock.tsx

```diff
@@ -42,25 +42,37 @@ export interface OneOfBlockPreviewState extends PreviewStateInterface {
     };
 }
 
+type ActiveType<Config extends boolean> = Config extends false ? { activeType: string } : { activeType?: string };
+
+export type OneOfBlockOutput<Config extends boolean> = {
+    attachedBlocks: {
+        type: string;
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        props: any;
+    }[];
+} & ActiveType<Config>;
+
 type BlockType = string;
-export interface CreateOneOfBlockOptions {
+export interface CreateOneOfBlockOptions<T extends boolean> {
     name: string;
     displayName?: React.ReactNode;
     supportedBlocks: Record<BlockType, BlockInterface>;
     category?: BlockCategory;
     variant?: "select" | "radio" | "toggle";
-    allowEmpty?: boolean;
+    allowEmpty?: T;
 }
 
-export const createOneOfBlock = ({
+export const createOneOfBlock = <T extends boolean = boolean>({
     supportedBlocks,
     name,
     displayName = "Switch",
     category = BlockCategory.Other,
     variant = "select",
-    allowEmpty = true,
-}: // eslint-disable-next-line @typescript-eslint/no-explicit-any
-CreateOneOfBlockOptions): BlockInterface<OneOfBlockFragment, OneOfBlockState, any, OneOfBlockPreviewState> => {
+    allowEmpty,
+}: CreateOneOfBlockOptions<T>): BlockInterface<OneOfBlockFragment, OneOfBlockState, OneOfBlockOutput<T>, OneOfBlockPreviewState> => {
+    // allowEmpty can't have a default type because it's typed by a generic
+    const internalAllowEmpty = (allowEmpty ?? true) satisfies boolean;
```

Comment:
Instead of `internalAllowEmpty` you could use `...restOptions` and `restOptions.allowEmpty ?? true`, which would remove the (to me) confusing additional variable.

Or you could rename the passed variable to `{ allowEmpty: passedAllowEmpty } `, which may be even more "beautiful", and we used this pattern in other cases

Reply:
I opted for the second approach